### ISSUE TITLE
Feature request: custom filter

### ISSUE THREAD

**Issue by wellloaded** [Jackett/Jackett#4749]
It would be nice if the user could define custom keywords to be added automatically (selectable custom defined option) to the search or even the filter, without having to type this is every time. e.g. 1080p

**Comment by wellloaded** [comment]
As a follow up to better explana what this ticket meant to say:

It would be very nice to be able to define manually custom display filters. So that they can be enabled/disabled with a flag.

e.g. in the search I would look for a movie title where in the display filter rather than type every time manually my preference I would ave one or more pre-defined keywords that I can check/uncheck as needed e.g. I would have one called "1080p" and another one called "x265". 
Beside enabling the filtering field to accept some basic "OR" function would be awesome. example: (5.1|6CH)

Thanks!

**Comment by RoloSoze** [comment]
@garfield69 @ngosang I like this suggestion. How doable is it?

**Comment by ngosang** [comment]
Everything is doable with enough time. I like this feature but the current Web UI codebase is in really bad shape and I don't want to make changes. It's really easy to break things.

**Comment by RoloSoze** [comment]
> 
> 
> Everything is doable with enough time. I like this feature but the current Web UI codebase is in really bad shape and I don't want to make changes. It's really easy to break things.

In what language is the Web UI written? And in general, other than C# & YAML, what skills and languages are required if people want to help out with Jackett? @cadatoiva @garfield69

**Comment by ngosang** [comment]
JS + HTML + CSS is the standard. The problem aren't the languages, it is bad designed. All the code is just in 2 files with more than 1000 lines each one mixing the 3 languages. No tests, ugly design, outdated dependencies... It has to be done from scratch but we don't have front developers. I can do the job but I don't want to maintain that code forever. Maybe I will do something but no promises.

**Comment by RoloSoze** [comment]
@ngosang I was actually just asking in general, not specifically for this particular feature. You guys are overwhelmed with issues, so I was thinking about posting in a bunch of private tracker forums to see if anyone has the willingness & the time to join you guys in the development team.

Thoughts? @cadatoiva @garfield69 

Also sorry for going off-topic. Feel free to move these messages elsewhere to continue this discussion.

**Comment by ngosang** [comment]
> I was actually just asking in general

Me too.

**Comment by cadatoiva** [comment]
@RoloSoze, It would be very appreciated if your recruitment efforts looked for frontend web developers in general. We really need a major overhaul to the whole UI system and aren't looking to salvage much from what's already there (except the API call/response system for now, to reduce our back-end load until we're ready to make changes). Since the rest of the project is in C#, and as C# is my strongest language, my bias is towards something ASP.Net adjacent or some other C#/NuGet based solution. I know @ngosang has experience with NPM, Typescript, React, Redux, and Ant, so that's another possibility. Those options ensure an existing member of the active team has the ability to participate in the process of the redesign in what ways we can, whether testing, directly contributing, maintaining, reviewing, etc.

Whatever we end up choosing, it needs to be fairly mainstream and maintainable, thus able to withstand changing hands considering the amount of churn the Jackett dev team has seen in our 5 years here.

**Comment by RoloSoze** [comment]
> It would be very appreciated if your recruitment efforts looked for frontend web developers in general.

Got it, but shouldn't I also include a call for C# & YAML developers for the backend stuff? Just in case I get lucky with a couple of people who may want to help.

**Comment by cadatoiva** [comment]
> Got it, but shouldn't I also include a call for C# & YAML developers for the backend stuff? Just in case I get lucky with a couple of people who may want to help.

Of course, I was just referring to the web ui specifically, to answer your question:

>And in general, other than C# & YAML, what skills and languages are required if people want to help out with Jackett?

The only other skill is Web front end UI/UX, and what we're looking for there and why is outlined above.

**Comment by RoloSoze** [comment]
@cadatoiva Sounds good. If we get lucky, and someone DMs me about wanting to help out, who should I direct them to?

**Comment by cadatoiva** [comment]
@RoloSoze, direct them to #8180 and we can take care of divy them up appropriately.
